File: DAA.Data/ArchivalEntity.cs
using System;
using System.Collections.Generic;

namespace DAA.Data
{
    public partial class ArchivalEntity
    {
        public ArchivalEntity()
        {
            DigitalObjects = new HashSet<DigitalObject>();
            Documents = new HashSet<Document>();
            Processes = new HashSet<Process>();
        }

        public int Id { get; set; }
        public int ArchiveId { get; set; }
        public DateTime? CreatedOn { get; set; }
        public Guid? CreatedBy { get; set; }
        public DateTime? UpdatedOn { get; set; }
        public Guid? UpdatedBy { get; set; }
        public bool Deleted { get; set; }
        public DateTime? DeletedOn { get; set; }
        public Guid? DeletedBy { get; set; }
        public int? ExternalIdentifier { get; set; }
        public bool HasExternalSource { get; set; }
        public DateTime? ExternalSourceUpdatedOn { get; set; }
        public string? Number { get; set; }
        public string? Title { get; set; }
        public string? DescriptionLevelCode { get; set; }
        public string? StatusCode { get; set; }
        public bool HasNoChronologicalScope { get; set; }
        public int? StartDateYear { get; set; }
        public int? StartDateMonth { get; set; }
        public int? StartDateDay { get; set; }
        public int? EndDateYear { get; set; }
        public int? EndDateMonth { get; set; }
        public int? EndDateDay { get; set; }
        public string? ApproxmateChronologicalScope { get; set; }
        public string? Author { get; set; }
        public string? Location { get; set; }
        public long? Bytes { get; set; }
        public int? SheetCount { get; set; }
        public int? TapeCount { get; set; }
        public int? MicrofilmCount { get; set; }
        public int? FrameCount { get; set; }
        public int? VideoTapeCount { get; set; }
        public int? DigitalDeviceCount { get; set; }
        public string? OtherMetrics { get; set; }
        public string? SizeCm { get; set; }
        public string? Scaling { get; set; }
        public string? Description { get; set; }
        public string? DocumentsAccessDescription { get; set; }
        public string? Features { get; set; }
        public string? Condition { get; set; }
        public int? MicrofilmedCopyCount { get; set; }
        public int? DigitizedCopyCount { get; set; }
        public int? PaperCopyCount { get; set; }
        public int? NegativeFrameCount { get; set; }
        public int? PositiveFrameCount { get; set; }
        public string? OtherCopyCount { get; set; }
        public string? Notes { get; set; }
        public long? EnrolledBytes { get; set; }
        public int? EnrolledDocumentCount { get; set; }
        public double? EnrolledLinearMeters { get; set; }
        public long? DeductedBytes { get; set; }
        public int? DeductedDocumentCount { get; set; }
        public double? DeductedLinearMeters { get; set; }
        public Guid SystemIdentifier { get; set; }
        public Guid FundSystemIdentifier { get; set; }
        public Guid InventorySystemIdentifier { get; set; }
        public bool IsSuspended { get; set; }
        public int? AvailabilityStatusCode { get; set; }
        public int? NumberNumeric { get; set; }
        public bool IsImported { get; set; }
        public string? NumberArray { get; set; }
        public string? DescriptionAuthor { get; set; }
        public string? Cypher { get; set; }
        public int? TextDocsCount { get; set; }
        public int? GraphicalDocsCount { get; set; }
        public string? Phase { get; set; }
        public string? Part { get; set; }
        public string? Stage { get; set; }
        public string? OtherLanguage { get; set; }
        public string? ClassificationSchemeIndex { get; set; }

        public virtual Archive Archive { get; set; } = null!;
        public virtual AvailabilityStatus? AvailabilityStatusCodeNavigation { get; set; }
        public virtual AspNetUser? CreatedByNavigation { get; set; }
        public virtual AspNetUser? DeletedByNavigation { get; set; }
        public virtual ArchivalEntityDescriptionLevel? DescriptionLevelCodeNavigation { get; set; }
        public virtual Fund FundSystemIdentifierNavigation { get; set; } = null!;
        public virtual Inventory InventorySystemIdentifierNavigation { get; set; } = null!;
        public virtual Status? StatusCodeNavigation { get; set; }
        public virtual AspNetUser? UpdatedByNavigation { get; set; }
        public virtual ICollection<DigitalObject> DigitalObjects { get; set; }
        public virtual ICollection<Document> Documents { get; set; }
        public virtual ICollection<Process> Processes { get; set; }
    }
}

File: DAA.Web.Intranet/client/src/models/archivalEntity.ts
import { IArchivalEntity, IArchivalEntityDraft } from "@/interfaces/archivalEntity";
//import { GridOptions } from "@/models/grid";

export class ArchivalEntity implements IArchivalEntity {
  constructor(obj?: IArchivalEntity) {
    Object.assign(this, obj);
  }

    id?: number;
    systemIdentifier?: string;
    isDraft?: boolean;
    archiveId?: number;
    fundDraftId?: number;
    fundSystemIdentifier?: string;
    fundExternalIdentifier?: number;
    fundHasExternalSource?: boolean;
    inventoryDraftId?: number;
    inventorySystemIdentifier?: string;
    inventoryExternalIdentifier?: number;
    inventoryHasExternalSource?: boolean;
    hasExternalSource?: boolean;
    externalIdentifier?: number;
    number?: string;
    numberNumeric?: number;
    descriptionLevelCode?: string;
    statusCode?: string;
    availabilityStatusCode?: number;
    creationMethodCodes?: Array<string>;
    languageCodes?: Array<string>;
    originalityCodes?: Array<string>;
    hasNoChronologicalScope?: boolean;
    startDateYear?: number;
    startDateMonth?: number;
    startDateDay?: number;
    endDateYear?: number;
    endDateMonth?: number;
    endDateDay?: number;
    approxmateChronologicalScope?: string;
    bytes?: number;
    linearMeters?: number;
    otherMetrics?: string;
    archivalEntityCount?: number;
    documentCount?: number;
    boxCount?: number;
    rollCount?: number;
    fundCreatorTitleHistory?: string;
    fundCreatorActivityHistory?: string;
    fundCreatorBiographicalHistory?: string;
    documentsProvider?: string;
    documentsDescription?: string;
    documentsAccessDescription?: string;
    history?: string;
    notes?: string;
    audioDocumentArchivalEntityCount?: number;
    photoDocumentArchivalEntityCount?: number;
    videoDocumentArchivalEntityCount?: number;
    digitalDocumentArchivalEntityCount?: number;
    classificationScheme?:string;
    abbreviationList?: string;
    microfilmedArchivalEntityCount?: number;
    digitizedArchivalEntityCount?: number;
    negativeFrameCount?: number;
    positiveFrameCount?: number;
    // createdOn?: Date;
    // deletedOn?: Date;
    // updatedOn?: Date;
    // createdBy?: string;
    // deletedBy?: string;
    // updatedBy?: string;
    // deleted?: boolean;
    // createdByUserName?: string;
    // createdByDisplayName?: string;
    // updatedByUserName?: string;
    // updatedByDisplayName?: string;
    // deletedByUserName?: string;
    // deletedByDisplayName?: string;
    resultMessage?: string; //TODO Remove this shit

    descriptionAuthor?: string;
    cypher?: string;
    textDocsCount?: number;
    graphicalDocsCount?: number;
    phase?: string;
    part?: string;
    stage?: string
    otherLanguage?: string;
    numberArray?: string;
    classificationSchemeIndex?: string;
}

export class ArchivalEntityDraft extends ArchivalEntity implements IArchivalEntityDraft {
  constructor(obj?: IArchivalEntityDraft) {
    super(obj)
    Object.assign(this, obj);
  }

  isCurrent?: boolean;
  readOnly?: boolean;
  workflowTypeCode?: string;
  workflowTypeText?: string;
  workflowId?: number;
  workflowStepTypeCode?: string;
  workflowStepTypeText?: string;
  workflowStepId?: number;
}

export class ArchivalEntityShort {
  constructor(obj?: IArchivalEntity) {
    Object.assign(this, obj);
    this.calculatedIdentifier = this.systemIdentifier + '|' + this.id?.toString() + '|' + this.externalIdentifier?.toString();
  }

  id?: number;
  systemIdentifier?: string;
  hasExternalSource?: boolean;
  externalIdentifier?: number;
  calculatedIdentifier?: string;
  isDraft?: boolean;
  archiveId?: number;
  archiveCode?: number;
  archiveName?: string;
  fundDraftId?: number;
  fundSystemIdentifier?: string;
  fundHasExternalSource?: boolean;
  fundExternalIdentiier?: number;
  fundNumber?: string;
  inventoryDraftId?: number;
  inventorySystemIdentifier?: string;
  inventoryHasExternalSource?: boolean;
  inventoryExternalIdentiier?: number;
  inventoryNumber?: string;
  number?: string;
  title?: string;
  approximateChronologicalScope?: string;
  descriptionLevelCode?: string;
  descriptionLevelText?: string;
  availabilityStatusCode?: number;
  availabilityStatusText?: string;
  statusCode?: string;
  statusText?: string;
  calculatedTitle?: string;
  commonId?: string; //TODO Remove
  name?: string; //TODO Remove
}

/* eslint-disable @typescript-eslint/no-explicit-any */
export class SearchedArchiveEntityRequestModel {
  constructor(obj?: any) {
    Object.assign(this, obj);
  }

  inventoryInternaIdentifier?: number;
  inventoryExternalIdentifier?: number;
  hasInventoryExternalSource?: boolean;
  searchText?: string;
}

export class DocumentAncestorsData {
  constructor(obj?: any) {
    Object.assign(this, obj);
  }

  archiveName?: string;
  archiveCode?: number;
  fundExternalIdentifier?: number;
  fundNumber?: string;
  inventoryExternalIdentifier?: number;
  inventoryNumber?: number;
  archiveEntityNumber?: string;
}
